src/coordinatorapi.cpp

```diff
@@ -501,7 +501,16 @@ bool waitForBarrier(const string& barrier,
     return false;
   }
 
```

Comment:
The new code continues to make the protocol of the coordinator incrementally more complicated.

Can you add a comment here or in `dmtcp_coordinator.cpp` that describes the state changes of this protocol?